### PR TITLE
Implement redesigned header

### PR DESCRIPTION
**Pull request by francoislehoux-okta**
## Description:
- **What's changed?** 
- Adds redesigned header, hidden behind `REDESIGNED` flag

## Netlify Build
[Click here](https://5fda3a22185e7b0cb3548f6e--reverent-murdock-829d24.netlify.app/pricing/)

## Screenshots:
Screen Shot 2020-12-16 at 12 26 05 PM
Screen Shot 2020-12-16 at 12 26 18 PM
Screen Shot 2020-12-16 at 12 26 24 PM

### Resolves:
* [OKTA-352704](https://oktainc.atlassian.net/browse/OKTA-352704)

### CODE REVIEW

File: packages/@okta/vuepress-site/.vuepress/public/img/icons/icon--menu-white.svg [hunk 1]
```diff
@@ -0,0 +1 @@
+<svg width="19" height="15" fill="none" xmlns="http://www.w3.org/2000/svg"><path stroke="#fff" stroke-width="1.75" stroke-linecap="round" stroke-linejoin="round" d="M1.078 1.034h16.551M1.078 7.579h16.551M1.078 14.125h16.551"/></svg>
```
Comment (shawyu-okta) [inline packages/@okta/vuepress-site/.vuepress/public/img/icons/icon--menu-white.svg:1]: I think it's OK either way probably, but could we avoid loading copies of the same SVG by setting the stroke color via CSS

File: packages/@okta/vuepress-theme-prose/components/Header.redesign.vue [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        <img v-if="searchOpened" src="/img/icons/icon--search-cherry.svg" />
+        <img v-else src="/img/icons/icon--search-white.svg" />
+      </div>
+      <div class="mobile--toggle" @click="toggleMenu()">
+        <img v-if="menuOpened" src="/img/icons/icon--menu-cherry.svg" />
+        <img v-else src="/img/icons/icon--menu-white.svg" />
+      </div>
+    </div>
+  </header>
+</template>
```
Comment (shawyu-okta) [inline packages/@okta/vuepress-theme-prose/components/Header.redesign.vue:40]: This template is super pleasing to read

File: packages/@okta/vuepress-theme-prose/components/Search.redesign.vue [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      <div class="CoveoAnalytics"></div>
+    </div>
+    <img class="input--icon" src="/img/icons/search.svg" />
+  </div>
+</template>
+
+<script>
+export default {
+  mounted() {
+    // Mount Coveo after page load
```
Comment (shawyu-okta) [inline packages/@okta/vuepress-theme-prose/components/Search.redesign.vue:14]: Ah, is this why I see the search seem to appear a moment after the rest of the page renders?

File: packages/@okta/vuepress-theme-prose/index.js [hunk 4]
```diff
@@ -14,9 +14,10 @@ module.exports = (options, ctx) => {
       $page.fullHeaders = resolveHeaders($page);
 
       // Set to true to enable the redesign in development
-      const redesign = process.env.REDESIGN || false;
+      // const redesign = process.env.REDESIGN || false;
       // Respect the redesign flag unless this is a production build
-      $page.redesign = !ctx.isProd && redesign;
+      // $page.redesign = !ctx.isProd && redesign;
+      $page.redesign = true;
```
Comment (shawyu-okta) [inline packages/@okta/vuepress-theme-prose/index.js:20]: 🚩 Just flagging with a comment to help remind us to revert this before merging

File: packages/@okta/vuepress-theme-prose/assets/css/layouts/_layout.scss [hunk 5]
```diff
@@ -53,13 +53,18 @@ body {
   background-color: cv('brand');
 }
 
-.page-body {
+.page-body:not(.redesign) {
```
Comment (shawyu-okta) [inline packages/@okta/vuepress-theme-prose/assets/css/layouts/_layout.scss:56]: I'm starting to wonder if we should set an `id` or `class` on the `body` element that we can globally scope our overrides on.

File: packages/@okta/vuepress-theme-prose/components/Search.redesign.vue (hunk 3 above)
Comment (francoislehoux-okta) [reply packages/@okta/vuepress-theme-prose/components/Search.redesign.vue:14]: yup

File: packages/@okta/vuepress-site/.vuepress/public/img/icons/icon--menu-white.svg (hunk 1 above)
Comment (francoislehoux-okta) [reply packages/@okta/vuepress-site/.vuepress/public/img/icons/icon--menu-white.svg:1]: Pretty sure you can only do this with inline SVGs

File: packages/@okta/vuepress-theme-prose/index.js (hunk 4 above)
Comment (francoislehoux-okta) [reply packages/@okta/vuepress-theme-prose/index.js]: Removed

File: packages/@okta/vuepress-site/.vuepress/public/img/icons/icon--menu-white.svg (hunk 1 above)
Comment (shawyu-okta) [reply packages/@okta/vuepress-site/.vuepress/public/img/icons/icon--menu-white.svg:1]: Oh, it's used as a background image. 😞

**Review by shawyu-okta — APPROVED** [review]
lgtm

**Pull request merged**